tools/llvm-spirv/llvm-spirv.cpp

```diff
@@ -202,6 +202,12 @@ static cl::opt<SPIRV::DebugInfoEIS> DebugEIS(
                    "extended instruction set. This version of SPIR-V debug "
                    "info format is compatible with the SPIRV-Tools")));
 
+static cl::opt<bool> SPIRVReplaceLLVMFmulAddWithOpenCLMad(
+    "spirv-replace-fmuladd-with-ocl-mad",
+    cl::desc("Allow replacement of llvm.fmuladd.* intrinsic with OpenCL mad "
+             "instruction from OpenCL extended instruction set"),
+    cl::init(true));
```

Comment:
How likely is it that similar user-requested replacements will be added for other intrinsics? If such plans (or such bits of code) exist already, would a single "joint" option like `-spirv-replace-intrinsics-with-ocl-ext` be preferable?

Reply:
> How likely is it that similar user-requested replacements will be added for other intrinsics?

This patch is more like a workaround for the lack of fine-grained fp-contraction support in SPIR-V spec, comparing it to OpenCL spec and LLVM IR representation. So, in general I wouldn't expect this to happen often, but still there might be a few more similar situations (right now I can't thing of any)

> If such plans (or such bits of code) exist already, would a single "joint" option like `-spirv-replace-intrinsics-with-ocl-ext` be preferable?

Since this is more like a workaround and I'm not aware of any similar things I guess it would be better to go with this particular option for `fmuladd`